frontend/packages/rhoas-plugin/src/utils/k8sUtils.ts

```diff
@@ -0,0 +1,52 @@
+import { k8sWatch } from '@console/internal/module/k8s';
+
+/**
+ * Use k8sWatch to wait for a resource to get into an expected condition.
+ * Watches for resource by kind, namespace and optional name.
+ * Promise resolves to a new resource version or rejects with a timeout.
+ * @param {K8sKind} kind
+ * @param {K8sResourceKind} resource
+ * @param {(kind: K8sResourceKind) => boolean} checkCondition
+ * @param {number} timeoutInMs
+ * @returns {Promise<K8sResourceKind>}
+ */
+export const k8sWaitForUpdate = (kind, resource, checkCondition, timeoutInMs) => {
```

Comment:
update core util instead of making a copy

Reply:
This comment was marked as resolved. We also discussed this on our call.
However no change has been made.